SRC/bank_01f.asm

```diff
@@ -0,0 +1,578 @@
+
+mapCollectionSubset:
+	dw bank_09_maps, bank_0A_maps, bank_0B_maps, bank_0C_maps, bank_0D_maps, bank_0E_maps, bank_0F_maps
+
+pauseMapTable:
+	dw landing
+	dw fissure
+	dw temple
+	dw lair1
+	dw grotto
+	dw lair2
+	dw tomb 
+	dw lair3
+	dw chasm
+	dw tower
+	dw abyss
+	dw omega
+	dw hive
+	dw escape
+
+m2maps_farLoadMapTiles:
+		; Get screen index from coordinates
+		ldh a, [hSamusYScreen]
+		rl a
+		rl a
+		rl a
+		rl a
+		and $f0
+		ld [samusMapY], a
+		ldh a, [hSamusXScreen]
+		and $0f
+		ld [samusMapX], a
+			;adjust index val based on direction uuuuuu
+			; Right
+			ld a, [doorScrollDirection]
+			cp a, $01
+			jr nz, .left
+				ld a, [samusMapX]
+				inc a
+				and a, $0f
+				ld [samusMapX], a
+				jr .none
+			.left:
+			ld a, [doorScrollDirection]
+			cp a, $02
+			jr nz, .up
+				ld a, [samusMapX]
+				dec a
+				and a, $0f
+				ld [samusMapX], a
+				jr .none
+			.up:
+			ld a, [doorScrollDirection]
+			cp a, $04
+			jr nz, .down
+				ld a, [samusMapY]
+				sub a, $10
+				and a, $f0
+				ld [samusMapY], a
+				jr .none						
+			.down:
+			ld a, [doorScrollDirection]
+			cp a, $08
+			jr nz, .none
+				ld a, [samusMapY]
+				add a, $10
+				and a, $f0
+				ld [samusMapY], a
+			.none:
+				ld a, [samusMapY]
+				ld b, a
+				ld a, [samusMapX]
+				add a, b
+				ld [mapCollectionIndex], a
+;					;debug hud view
+;					ld a, [mapCollectionIndex]
+;					ld [samusDispMissilesLow], a
+;					ld [samusCurMissilesLow], a
+	;set up map index based on currentLevelBank
+	ld a, [currentLevelBank]
+	sub mapTable_bankOffset
+;		;debug hud view
+;		ld [samusCurMissilesHigh], a
+;		ld [samusDispMissilesHigh], a
+	add a, a
+	ld e, a
+		;debug, and for later
+		ld [mapLevelBankIndexOffset], a
+	ld d, $00
+	ld hl, mapCollectionSubset
+	add hl, de
+		;now we have pointer to the proper map collection table to review (indexed by currentLevelBank)
+		;load the value at that submap table index to hl to find the proper submap
+		ld e, [hl]
+		inc hl
+		ld d, [hl]
+		ld h, d
+		ld l, e
+		;correct mapCollectionSubset index by bank pointer should be in HL now
+		;so now, enter table based on hSamusYScreen/hSamusXScreen
+		ld d, $00
+		ld a, [mapCollectionIndex]
+		ld e, a
+		add hl, de
+			ld a, l
+			ld [mapCollectionTableXY], a
+			ld a, h
+			ld [mapCollectionTableXY+1], a
+				;now we have pointer to the proper map (indexed by mapCollectionIndex)
+				;load it and prepare to read data from the map to Window VRAM targets
+				ld e, [hl]
+				ld d, [hl]
+				ld h, d
+				ld l, e
+
+	;now load value at HL addr and double it as pointer to submap to read from
+	ld a, l
+	add a, a
+	ld e, a
+	ld d, $00
+	ld hl, pauseMapTable
+	add hl, de	
+	;hl now points to map pointer
+	;
+	ld e, [hl]
+	inc hl
+	ld d, [hl]
+	ld h, d
+	ld l, e
+	;
+    ld de, vramDest_m2maps
+    ld b, $00
+    :
+        ld a, [hl+]
+        ld [de], a
+        inc de
+        ld a, [hl+]
+        ld [de], a
+        inc de
+        dec b
+		ld a, b
+		and $0f
+		cp a, $06
```

Comment:
Maybe change 6 to `$10 - (map_width/2)` where `map_width = 20 ; Screen width (160px) / tile width 8px)